itest/lnd_channel_backup_test.go

```diff
@@ -1672,3 +1954,383 @@ func assertDLPExecuted(ht *lntest.HarnessTest,
+		ht.MineEmptyBlocks(int(defaultCSV - blocksMined))
+
+		// We'll now mine the remaining blocks to prompt Dave to sweep
+		// his CLTV-constrained output.
+		resp := dave.RPC.PendingChannels()
+		blocksTilMaturity :=
+			resp.PendingForceClosingChannels[0].BlocksTilMaturity
+		require.Positive(ht, blocksTilMaturity)
+
+		ht.MineEmptyBlocks(int(blocksTilMaturity))
+
+		// Dave should have two sweep requests - one for his commit
+		// output and the other for his anchor.
+		ht.AssertNumPendingSweeps(dave, 2)
+
+		// Mine a block to trigger the sweep.
+		ht.MineEmptyBlocks(1)
+		ht.MineBlocksAndAssertNumTxes(1, 1)
+
+		// Now Dave should consider the channel fully closed.
+		ht.AssertNumPendingForceClose(dave, 0)
+	} else {
+		// The funds are timelocked, so Dave has one sweep.
+		ht.AssertNumPendingSweeps(dave, 1)
+
+		// Mine one block to trigger the sweeper to sweep.
+		ht.MineEmptyBlocks(1)
+		blocksMined++
+
+		// After Dave's output matures, he should also reclaim his
+		// funds.
+		//
+		// The commit sweep resolver publishes the sweep tx at
+		// defaultCSV-1 and we already have blocks mined after the
+		// commitmment was published, so take that into account.
+		ht.MineEmptyBlocks(int(defaultCSV - blocksMined))
+
+		// Mine one block to trigger the sweeper to sweep.
+		ht.MineEmptyBlocks(1)
+
+		// Dave's sweeper swept the outputs of force close tx.
+		if lntest.CommitTypeHasAnchors(commitType) {
+			ht.AssertNumPendingSweeps(dave, 2)
+		} else {
+			ht.AssertNumPendingSweeps(dave, 1)
+		}
+
+		// Assert the sweeping tx is mined.
+		ht.MineBlocksAndAssertNumTxes(1, 1)
+
+		// Funds are now in Dave's wallet.
+		ht.AssertNumPendingForceClose(dave, 0)
+	}
+
+	// We query Dave's balance to make sure it increased after the channel
+	// closed. This checks that he was able to sweep the funds he had in
+	// the channel.
+	daveBalResp := dave.RPC.WalletBalance()
+	daveBalance := daveBalResp.ConfirmedBalance
+	require.Greater(ht, daveBalance, daveStartingBalance,
+		"balance not increased")
+
+	ht.AssertNodeNumChannels(dave, 0)
+}
+
+// assertFundsRecoveredStartCarol asserts that funds are recovered after Dave
+// running scbforceclose procedure. Force close transaction is created from
+// Dave's SCB backup and broadcasted by the caller of this function. Carol is
+// started after the force close transaction is mined.
+func assertFundsRecoveredStartCarol(ht *lntest.HarnessTest,
+	carol *node.HarnessNode, carolStartingBalance int64,
+	dave *node.HarnessNode, daveStartingBalance int64,
+	commitType lnrpc.CommitmentType, startCarol func() error) {
+
+	ht.Helper()
+
+	// Increase the fee estimate so that the following force close tx will
+	// be cpfp'ed.
+	ht.SetFeeEstimate(30000)
+
+	// Upon reconnection, the nodes should detect that Dave is out of sync.
+	// Carol should force close the channel using her latest commitment.
+	// For scbforceclose case: we detect the manually broadcasts Dave's tx.
```

Comment:
According to the test it looks to me that  `assertFundsRecoveredStartCarol ` is only called in the scbforceclose case ?

Reply:
Fixed. It was a relict of my original approach trying to have a single "assert" function for all cases. The function became too branched and complicated and I split it into separate "assert" functions, but forgot to update this comment.